### PR TITLE
[BUG] fixing bug for `_split` (accidental removal of `pandas.Index` support)

### LINKED ISSUE #1580: [BUG] model_selection/split does not support y being an index since 0.8.1
**Describe the bug**
#1456 introduced a bug. If y is an Index, the code leads to an unassigned local variable
```
UnboundLocalError: local variable 'y_index' referenced before assignment
```

**To Reproduce**
```python
from sktime.forecasting.model_selection import SingleWindowSplitter
import pandas as pd
next(SingleWindowSplitter(fh=1).split(pd.PeriodIndex(year=[2000, 2002], quarter=[1, 3])))
```

**Expected behavior**
split function should accept an Index as argument

**Versions**
0.8.1

### PR DESCRIPTION
**Pull request by fkiraly**
This fixes #1580, by allowing `pandas.Index` to be passed to `_check_y` again.

### CODE REVIEW

File: sktime/forecasting/model_selection/_split.py [hunk 1]
```diff
@@ -97,11 +98,34 @@ def has_changed(k, v):
...
+    """
     if isinstance(y, (pd.Series, pd.DataFrame)):
         y_index = y.index
-    if isinstance(y, np.ndarray):
+    elif isinstance(y, np.ndarray):
         y_index = pd.Index(y)
+    elif isinstance(y, pd.Index):
+        y_index = y
+    else:
+        raise ValueError(
```
Comment (hoesler4dh) [inline sktime/forecasting/model_selection/_split.py:126]: Isn't a TypeError more suitable here?

Comment (fkiraly) [reply sktime/forecasting/model_selection/_split.py:126]: yes, of course.

Comment (fkiraly) [reply sktime/forecasting/model_selection/_split.py]: fixed

**Comment by fkiraly** [comment]
@mloning, @aiwalter, can you please review? This is a simple bugfix.

File: sktime/forecasting/model_selection/_split.py [hunk 2]
```diff
@@ -97,11 +98,35 @@ def has_changed(k, v):
 
 
 def _check_y(y):
-    """Check input to `split` function."""
+    """Coerce input to `split` function.
+
+    Parameters
+    ----------
+    y : pd.Series, pd.DataFrame, np.ndarray, or pd.Index
```
Comment (chrisholder) [inline sktime/forecasting/model_selection/_split.py:105]: Add a description

File: sktime/forecasting/model_selection/_split.py [hunk 3]
```diff
@@ -97,11 +98,35 @@ def has_changed(k, v):
...
+    check_time_index(y_index), where y_index is as follows:
+        if y is pd.Series, pd.DataFrame, y_index = y.index
+        if y is pd.Index, y_index = y itself
+        if y is np.ndarray, y_index =  pd.Index(y)
+
+    Raises
+    ------
+    TypeError if y is not of one of the expected types
+    NotImplementedError if y_index is not a supported sktime index type
+    ValueError if y_index is not monotonous
```
Comment (chrisholder) [inline sktime/forecasting/model_selection/_split.py:118]:
```
TypeError 
    if y is not of one of the expected types
NotImplementedError 
    if y_index is not a supported sktime index type
ValueError 
    if y_index is not monotonous
```
I know you've only got one case for each but my ocd is getting the better of me here.

File: sktime/forecasting/model_selection/_split.py (hunk 2 above)
Comment (fkiraly) [reply sktime/forecasting/model_selection/_split.py:105]: done

File: sktime/forecasting/model_selection/_split.py (hunk 3 above)
Comment (fkiraly) [reply sktime/forecasting/model_selection/_split.py]: done

**Comment by fkiraly** [comment]
@chrisholder, comments addressed, please review

**Review by chrisholder — APPROVED** [review]

**Pull request merged**